framework/include/base/MooseApp.h

```diff
@@ -1322,7 +1328,7 @@ class MooseApp : public ConsoleStreamInterface,
   std::shared_ptr<Backup> _cached_backup;
 
   /// Execution flags for this App
-  const ExecFlagEnum & _execute_flags;
+  const ExecFlagEnum _execute_flags;
```

Comment:
Why is this change needed?

Reply:
This is one of the most critical changes in this patch. Without it, an application will see a different set of flags when it's first being setup versus when some of the post "add_multiapp" tasks run. This results in errors about unregistered flags during some of the processing we do in the warehouses to setup objects based on ExecFlags. 

While there may be better designs to this problem, I argue this is the better "workaround" for now. However, it might be prudent to put in a detailed comment about why this is like this. The specific error that I ran into was "Pronghorn" complaining that it didn't know about "XFEM_MARK", an exec flag registered in the "xfem" module used by Bison, but not Pronghorn. I tend to agree that Pronghorn doesn't need storage classes for "XFEM_MARK" and this intentional copy of the flags prevents that.

Reply:
Yea I think some of this information would be great as a comment to communicate the intent, otherwise another developer may come through here and change to a reference again, especially as we do not have testing anywhere (yet)

Reply:
Added